### ISSUE TITLE
Channel Source is borked due to stop channel injection

### ISSUE THREAD

**Issue by DirectXMan12** [kubernetes-sigs/controller-runtime#103]
The Channel source expects the stop channel to be injected before calling `Start`, but the stop channel on the controller manager will be nil until `ControllerManager#Start` is called.  Since most people will probably call `Controller#Watch` before `ControllerManager#Start`, this breaks the channel source.

I've got a patch to fix it in the works, but wanted to file this so that I don't forget.

**Comment by AMecea** [comment]
@DirectXMan12 any news about this?

**Comment by DirectXMan12** [comment]
no, unfortunately I became busy with some other stuff, and it's going to take a bit of thinking about how to refactor the internals to make it work.  I'll take another stab at it next week.

**Comment by hzamani** [comment]
One solution is to add a `Initialize` method to `Reconciler` interface and call that in `Controller#Start`, then move all `Controller#Watch` calls to this method. This method can be used for other initialization stuff what require `ControllerManager` to be started (for example operations that require a synced catch).
